test/unit/test_generate_checksum.py

```diff
@@ -0,0 +1,132 @@
+# pylint: disable=C0116
+# pylint: disable=C0114
+from unittest.mock import mock_open
+import mock
+import pytest
+
+from generate_checksum.checksum import (
+    validate_checksums,
+    compare_hash,
+    write_checksum_file,
+    generate_md5,
+    generate_sha512
+)
+
+@mock.patch('generate_checksum.checksum.Path', autospec=True)
+def test__validate_checksums__validate_checksums_passes_with_no_checksum_file(mock_path):
+    mock_path.suffix.return_value = ''
+    mock_path.with_suffix.return_value = mock_path
+    mock_path.exists.return_value = False
+
+    validate_checksums(mock_path)
+
+@mock.patch('generate_checksum.checksum.compare_hash')
+@mock.patch('generate_checksum.checksum.Path', autospec=True)
+def test__validate_checksums__validate_checksum_passes(mock_path, mock_compare_hash):
+    mock_path.suffix.return_value = ''
+    mock_path.with_suffix.return_value = mock_path
+    mock_path.exists.return_value = True
+
+    mock_compare_hash.return_value =  True
+
+    validate_checksums(mock_path)
+
+@mock.patch('generate_checksum.checksum.compare_hash')
+@mock.patch('generate_checksum.checksum.Path', autospec=True)
+def test__validate_checksums__error_raised_when_comparison_fails(mock_path, mock_compare_hash):
+    mock_path.suffix.return_value = ''
+    mock_path.with_suffix.return_value = mock_path
+    mock_path.exists.return_value = True
+
+    mock_compare_hash.return_value =  False
+
+    with pytest.raises(IOError):
+        validate_checksums(mock_path)
+
+@mock.patch('generate_checksum.checksum.generate_md5')
+@mock.patch('generate_checksum.checksum.Path', autospec=True)
+def test__compare_hash__compares_correct_md5_checksums(mock_path, mock_generate_md5):
+    checksum_to_compare = '123456abcd'
+    hash_type = 'md5'
+    mock_path.read_text.return_value = f'{checksum_to_compare} fname'
+
+    mock_generate_md5.return_value = checksum_to_compare
+
+    assert compare_hash(hash_type, mock_path, mock_path)
+
+@mock.patch('generate_checksum.checksum.generate_sha512')
+@mock.patch('generate_checksum.checksum.Path', autospec=True)
+def test__compare_hash__compares_correct_sha512_checksums(mock_path, mock_generate_sha512):
```

Comment:
You can use `pytest.mark.parametrize` to reuse tests with different inputs. Here's [an example](https://github.com/uclahs-cds/project-CompoundAssayPlatform/blob/main/CAP/test/unit/app/config/test_Config.py#L10-L22).